File: dcp-solutions/2024-09-27/solution.php
```php
function maxSubarray($nums, $k) {
    $n = count($nums);
    if ($n == 0 || $k == 1) {
        return;
    }

    $max_values = array_fill(0, $n - $k + 1, 0);

    for ($i = 0; $i < $n - $k + 1; $i++) {
        $current_max = PHP_INT_MIN;
        for ($j = $i; $j < $i + $k; $j++) {
            $current_max = max($current_max, $nums[$j]);
        }
        $max_values[$i] = $current_max;

        print($current_max . " ");
    }

    print("\n");
}

```

File: dcp-solutions/2024-09-27/solution.ex
```elixir
def max_subarray(nums, k) do
  # initialize variables
  prefix_sum = Enum.reduce(0..k-1, 0, fn(i, acc) -> Enum.sum(Enum.at(nums, i..i+k-1)) + acc end)
  # Kadane's algorithm with prefix sum
  Enum.each_over_index(k..Enum.count(nums)-1, fn(i, j) ->
    # subtract the first element of previous window
    new_sum = Enum.sum(Enum.at(nums, j+1..j+k))
    new_prefix_sum = new_sum - prefix_sum
    if new_prefix_sum > prefix_sum do
      prefix_sum = new_prefix_sum
    end
  end)
  # print result
  for i <- k..Enum.count(nums), into: []
  end
end
```